v2/api/cognitiveservices/customizations/account_extension.go

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package customizations
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/cognitiveservices/armcognitiveservices"
+	"github.com/go-logr/logr"
+	"github.com/rotisserie/eris"
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+
+	storage "github.com/Azure/azure-service-operator/v2/api/cognitiveservices/v1api20250601/storage"
+	"github.com/Azure/azure-service-operator/v2/internal/genericarmclient"
+	"github.com/Azure/azure-service-operator/v2/internal/set"
+	"github.com/Azure/azure-service-operator/v2/internal/util/to"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime/secrets"
+)
+
+const (
+	accountKey1 = "Key1"
+	accountKey2 = "Key2"
+	accountEp   = "Endpoint"
+	accountEps  = "Endpoints"
+)
+
+var _ genruntime.KubernetesSecretExporter = &AccountExtension{}
+
+func (ext *AccountExtension) ExportKubernetesSecrets(
+	ctx context.Context,
+	obj genruntime.MetaObject,
+	additionalSecrets set.Set[string],
+	armClient *genericarmclient.GenericClient,
+	log logr.Logger,
+) (*genruntime.KubernetesSecretExportResult, error) {
+	typedObj, ok := obj.(*storage.Account)
+
+	if !ok {
+		return nil, eris.Errorf("Cannot run on unknown resource type %T, expected *storage.Account", obj)
+	}
+
+	var _ conversion.Hub = typedObj
+
+	primarySecrets := secretsSpecified(typedObj)
+	requestedSecrets := set.Union(primarySecrets, additionalSecrets)
+
+	if len(requestedSecrets) == 0 {
+		log.V(1).Info("No secrets to retrieve as operatorSpec is empty")
+		return nil, nil
+	}
+
+	id, err := genruntime.GetAndParseResourceID(typedObj)
+	if err != nil {
+		return nil, err
+	}
+
+	keys := make(map[string]string)
+
+	if primarySecrets.Contains(accountKey1) || primarySecrets.Contains(accountKey2) {
+		subscription := id.SubscriptionID
+		acctClient, err := armcognitiveservices.NewAccountsClient(subscription, armClient.Creds(), armClient.ClientOptions())
+		if err != nil {
+			return nil, eris.Wrapf(err, "Failed to create new AccountsClient")
+		}
+
+		resp, err := acctClient.ListKeys(ctx, id.ResourceGroupName, typedObj.AzureName(), nil)
+		if err != nil {
+			return nil, eris.Wrapf(err, "Failed listing keys")
+		}
+
+		keys[accountKey1] = to.Value(resp.Key1)
+		keys[accountKey2] = to.Value(resp.Key2)
+	}
+
+	keys[accountEp] = getSafeString(typedObj.Status.Properties.Endpoint)
```

Comment:
minor: We have a helper for this: v2/internal/util/to/to.go
`Value[T any](v *T) T` -- use `to.Value(...)`

```suggestion
	keys[accountEp] = to.Value(typedObj.Status.Properties.Endpoint)
```